File: Programmers/약수의 개수와 덧셈.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int left, int right) {
    int answer = 0;

    for(int i=left;i<=right;i++){
        int tmp = 0;
        for(int j=1;j<=i;j++){
            if(i % j == 0){
                tmp++;
            }
        }
        if(tmp % 2 == 0){
            answer += i;
        }
        else
            answer -= i;
    }

    return answer;
}

File: Programmers/약수의 개수와 덧셈.java
class Solution {
    public int solution(int left, int right) {
        int answer = 0;
        while(left <= right)
        {
            int tmp = 0;
            for(int i = 1; i <= left; i++)
                if(left % i == 0)
                    tmp++;
            if(tmp % 2 == 0)
                answer += left;
            else
                answer -= left;
            left++;
        }
        return answer;
    }
}